cmd/syncthing/gui.go

```diff
@@ -1024,6 +1027,93 @@ func (s *apiService) getSystemLogTxt(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+type fileEntry struct {
+	name string
+	data []byte
+}
+
+func (s *apiService) getSupportBundle(w http.ResponseWriter, r *http.Request) {
+	var files []fileEntry
+
+	// Redacted configuration as a JSON
+	if jsonConfig, err := json.MarshalIndent(getRedactedConfig(s), "", "  "); err == nil {
+		files = append(files, fileEntry{name: "config.json", data: jsonConfig})
+	}
```

Comment:
Should we not do something in case of error, rather than skipping it completely? Warn or something atleast?

Reply:
I added error checks for all json marshalling, I'm not sure needed or not. Any suggestions?